src/gen-hb-version.py

```diff
@@ -32,5 +32,8 @@
 			.replace ("@HB_VERSION@", version)
 			.encode ())
 
-# copy it also to src/
-shutil.copyfile (OUTPUT, os.path.join (CURRENT_SOURCE_DIR, os.path.basename (OUTPUT)))
+# copy it also to the source tree, but only if it has changed
+with open(os.path.join (CURRENT_SOURCE_DIR, os.path.basename (OUTPUT)), "r", encoding='utf-8') as baseline:
```

Comment:
You are opening one file with `encoding='utf-8'` and the other not, this will cause trouble on Windows if a non-ASCII character appeared in these files. Better use the default, or even open the files in binary mode so Python does not temper with the content.

Reply:
The "encoding" parameter was lifted from the open earlier in the code. Is opening in binary mode going to cause a problem with crlf/lf mismatches?

Reply:
We open the file for writing in binary mode above, so it should be using the lf on all platforms already.

Reply:
Mode is now "rb", the encoding is gone, and the filename is now in a variable as per the other script.